packages/integration-tests/src/tests/api/oidc/token-exchange.test.ts

```diff
@@ -134,5 +137,119 @@ describe('Token Exchange', () => {
 
       await deleteApplication(thirdPartyApplication.id);
     });
+
+    it('should filter out non-oidc scopes', async () => {
+      const { subjectToken } = await createSubjectToken(userId);
+
+      const body = await oidcApi
+        .post('token', {
+          headers: formUrlEncodedHeaders,
+          body: new URLSearchParams({
+            client_id: applicationId,
+            grant_type: GrantType.TokenExchange,
+            subject_token: subjectToken,
+            subject_token_type: 'urn:ietf:params:oauth:token-type:access_token',
+            scope: [UserScope.Profile, 'non-oidc-scope'].join(' '),
+          }),
+        })
+        .json();
+
+      expect(body).toHaveProperty('access_token');
```

Comment:
better to decode jwt and assert claims. i [learned a lesson](https://github.com/logto-io/logto/pull/6170/files) about this.

Reply:
Good call, created an issue to track.